code/modules/antagonists/abductor/equipment/gland.dm

```diff
@@ -15,6 +15,16 @@
 	var/mind_control_uses = 1
 	var/mind_control_duration = 1800
 	var/active_mind_control = FALSE
+	var/gland_desc = " generic"
+
+/obj/item/organ/heart/gland/examine(mob/user)
+	.=..()
+	if(istype(user, /mob/living/carbon))
+		var/mob/living/carbon/M = user
+		if(istype(M.dna.species, /datum/species/abductor))
+			var/datum/species/abductor/S = M.dna.species
+			if(S.scientist)
+				to_chat(user, "You discern it to be a[gland_desc] gland.")
```

Comment:
Just have the glands have names like "slime gland", "regeneration gland", and then have them rename themselves permamently to something generic when inserted into a person.  Saves the specialcasing.

Reply:
That'd work, but I thought this way would make it clear that the scientist should be the one choosing the glands. The agent already has most of the actual abductor gameplay and powers in the form of stealthing and kidnapping, while the scientist just does the same surgery again and again and watches on camera.

Reply:
It's similar to how the tools force a scientist check to prevent the agent being overbearing and being the only useful member of the team.

Reply:
tbh, I felt like the Scientist was the really important role of the team, given if the Scientist dies, it's game over for the both of them.

Reply:
They're important, but that doesn't mean they have interesting stuff to do (barring brainwash, which people ignore most of the time anyway)

Reply:
I like this idea, but I have a better implementation idea.

Reply:
If the scientist had no other properties than wandering around on station and killing the agent when it died it'd be important, but it wouldn't have anything interesting to do. I feel the current situation is similar.

Reply:
Fair enough on the implementation, go for it.